OpenRA.Mods.Common/Traits/Air/AttackAircraft.cs

```diff
@@ -48,6 +51,21 @@ public AttackAircraft(Actor self, AttackAircraftInfo info)
 			aircraftInfo = self.Info.TraitInfo<AircraftInfo>();
 		}
 
+		public override bool TargetInFiringArc(Actor self, in Target target, WAngle facingTolerance)
+		{
+			if (facing == null)
+				return true;
+
+			var pos = self.CenterPosition;
+			var targetedPosition = GetTargetPosition(pos, target);
+			var delta = targetedPosition - pos;
+
+			if (delta.HorizontalLengthSquared == 0)
+				return true;
+
+			return Util.FacingWithinTolerance(facing.Facing, delta.Yaw + Info.FiringAngle, facingTolerance);
```

Comment:
And then we'd do something like the following here:
```suggestion
			foreach (var facing in Info.FacingOffsets)
				if (Util.FacingWithinTolerance(facing.Facing, delta.Yaw + facing, facingTolerance))
					return true;

			return false;
```

That could then cover your "both sides" use case.

Reply:
Good idea here but Milk concerns that the fire position muzzle should be correct on those offsets, which I think may need a refactor that move fire offset to "Attack" instead of "Armament".

Reply:
That might be a problem. I'm not sure if moving the offsets to Attack is a good idea, will have to think about that.

Reply:
Then we can just ignore it, it is still enough for many usecases.